aws/external/provider.go

```diff
@@ -320,6 +321,6 @@ type WithEC2MetadataRegion struct {
 }
 
 // GetRegion attempts to retreive the region from EC2 Metadata service.
-func (p WithEC2MetadataRegion) GetRegion() (string, error) {
-	return p.Client.Region()
+func (p WithEC2MetadataRegion) GetRegion(ctx context.Context) (string, error) {
+	return p.Client.Region(ctx)
```

Comment:
Adding `context` to this helper will prevent the `WithEC2MetadataRegion` provider from beign used because it will no longer satisfy the `RegionProvider` interface that is used by the config to generically get the region.

I suggest instead, refactor the `WithEC2MetadataRegion` to have a constructor function that takes a context, and client. Making the `Client` member of the struct unexported.

```go
type WithEC2MetadataRegion struct {
     ctx context.Context
     client *ec2metadata.Client
}
func NewWithEC2MetadataRegion(ctx context.Context, client *ec2metadata.Client) *WithEC2MetadataRegion {
//...
}
```

Usage
```go
ec2metaClient := ec2metadata.New(defaults.Config())

cfg, err := external.LoadDefaultAWSConfig(
    external.NewWithEC2MetadataRegion(ctx, ec2metaClient),
)
```

Reply:
updated in #461